File: sdk/typescript/src/examples/quickstart.ts
import { readFile } from "node:fs/promises";

import {
  CigarClient,
  bundleId,
  createIdempotencyKey,
  verifyBundle,
  type SemanticContextBundle,
} from "../index.js";

interface Fixture {
  readonly schema_version: string;
  readonly bundle: SemanticContextBundle;
  readonly expected_bundle_id: string;
}

const fixtureUrl = new URL("../../fixtures/semantic-bundle-v1.json", import.meta.url);
const fixture = JSON.parse(await readFile(fixtureUrl, "utf8")) as Fixture;
if (fixture.schema_version !== "cigar.sdk-semantic-bundle-fixture.v1") throw new Error("unsupported fixture");
verifyBundle(fixture.bundle);
const identity = bundleId(fixture.bundle);
if (identity !== fixture.expected_bundle_id) throw new Error("shared semantic bundle identity differs");

const baseUrl = process.env.CIGAR_URL;
if (baseUrl !== undefined) {
  const planId = process.env.CIGAR_PLAN_ID;
  if (planId === undefined) throw new Error("CIGAR_PLAN_ID is required with CIGAR_URL");
  const client = new CigarClient({
    baseUrl,
    allowInsecureLoopback: baseUrl.startsWith("http://"),
    ...(process.env.CIGAR_TOKEN === undefined ? {} : { bearerToken: process.env.CIGAR_TOKEN }),
  });
  await client.negotiate({ timeoutMs: 5_000 });
  const compiled = await client.compileContextBundle({
    payload: { plan_id: planId },
    idempotencyKey: createIdempotencyKey("quickstart"),
  });
  if (compiled.payload.bundle_id !== identity) throw new Error("daemon bundle identity differs from the shared fixture");
  const manifest = await client.getContextBundleManifest({ payload: { bundle_id: compiled.payload.bundle_id } });
  console.error(`verified daemon manifest ${manifest.payload.manifest_id}`);
}

console.log(identity);

File: sdk/python/examples/quickstart.py
"""Verify the shared identity and optionally execute the async daemon workflow."""

from __future__ import annotations

import asyncio
import json
import os
import sys
from importlib import resources  # nosemgrep: python.lang.compatibility.python37.python37-compatibility-importlib2

from cigar_sdk import AsyncCigarClient, CallOptions, TypedOperationRequest, bundle_id, create_idempotency_key, models
from cigar_sdk.digest import verify_bundle


async def main() -> None:
    source = resources.files("cigar_sdk.fixtures").joinpath("semantic-bundle-v1.json").read_text(encoding="utf-8")
    fixture = json.loads(source)
    verify_bundle(fixture["bundle"])
    identity = bundle_id(fixture["bundle"])
    if identity != fixture["expected_bundle_id"]:
        raise ValueError("shared semantic bundle identity differs")

    endpoint = os.environ.get("CIGAR_URL")
    if endpoint is not None:
        plan_id = os.environ.get("CIGAR_PLAN_ID")
        if plan_id is None:
            raise ValueError("CIGAR_PLAN_ID is required with CIGAR_URL")
        async with AsyncCigarClient(
            endpoint,
            bearer_token=os.environ.get("CIGAR_TOKEN"),
            allow_insecure_loopback=endpoint.startswith("http://"),
        ) as client:
            await client.negotiate(options=CallOptions(timeout=5.0))
            compiled = await client.compile_context_bundle(
                TypedOperationRequest(
                    models.CompileContextBundleRequest(plan_id=plan_id),
                    idempotency_key=create_idempotency_key("quickstart"),
                )
            )
            if compiled.payload.bundle_id != identity:
                raise ValueError("daemon bundle identity differs from the shared fixture")
            manifest = await client.get_context_bundle_manifest(
                TypedOperationRequest(models.BundleIdRequest(bundle_id=compiled.payload.bundle_id))
            )
            print(f"verified daemon manifest {manifest.payload.manifest_id}", file=sys.stderr)
    print(identity)


if __name__ == "__main__":
    asyncio.run(main())
